week5/main.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import FastAPI, File, UploadFile
+from fastapi.responses import StreamingResponse
+import pandas as pd
+import os
+from io import BytesIO, StringIO
+from serving.ml_decision_tree_sample.prepare_data import prepare_data_for_train, common_prepare, prepare_data_for_predict
+from serving.ml_decision_tree_sample.predictor import model_for_predict
+from serving.ml_decision_tree_sample.load_model import model_loader
+from serving.ml_decision_tree_sample.train import grid_train_model
+from serving.ml_decision_tree_sample.cleaner import clean_all
+from serving.config import model_file_name
+import uvicorn
+
+
+
+
+def request_csv_to_df(file: UploadFile) -> pd.DataFrame:
+    contents = file.file.read()
+    buffer = BytesIO(contents)
+    df = pd.read_csv(buffer)
+    buffer.close()
+    file.file.close()
+    return df
+
+
+def return_df_like_csv_file_in_response(df: pd.DataFrame):
+    stream = StringIO()
+    df.to_csv(stream, index=False)
+    response = StreamingResponse(iter([stream.getvalue()]), media_type="text/csv")
+    response.headers["Content-Disposition"] = "attachment; filename=predict.csv"
+    return response
+
+
+app = FastAPI()
+
+
+@app.get("/")
+def read_root():
+    return {"Hello": "please add \'/docs\' to ur url and u will be transferred to swagger page with list of request "
+                     "example url http://127.0.0.1:8000/docs"}
+
+
+@app.post("/train_model_and_push_to_wandb/")
+async def train_model_and_push_to_wandb(file: UploadFile = File(...)):
```

Comment:
should we specify returning scheme here?

Reply:
train request nothing to return